### PR TITLE
Authentication and Token Authorization

### PR DESCRIPTION
**Pull request by rathukugan**
- Account registration with email & password through /user endpoint (email/password validation happens here & we can add email verification here in the future)
- Login through /login endpoint with email & password, returns a token (JWT) - Current expiry time of 2h
- For any API requests (/post etc) send the token as a bearer token (https://www.getpostman.com/docs/v6/postman/sending_api_requests/authorization#bearer-token) to get authorization and get back your JSON output!

**Still need to do some more testing for corner cases, replace placeholders with env variables and push to heroku
but PR is ready for review.**

### CODE REVIEW

File: db_connection.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    MongoClient.connect(dbURI, function(err, database) {
+      _db = database.db('healthoutloud');
+      return callback(err);
+    });
+  },
+
+  getDb: function() {
+    return _db;
+  }
+};
```
Comment (JustCourt) [inline db_connection.js:18]: newline

File: index.js [hunk 2]
```diff
@@ -1,113 +1,124 @@
...
+	var user = new User(req.body.email);
+	//Validate email and password
+	if (!user.isValidEmail(req.body.email)) return res.status(400).send('Invalid email');
+	if (!user.isValidPassword(req.body.password)) return res.status(400).send('Invalid password');
+
+	//Check if email already exists
+  	db.collection('user').count({ email: req.body.email }, function (err, count){ 
+  		if (err) return console.log(err);
+
+    	if(count > 0) {
```
Comment (JustCourt) [inline index.js:45]: indenting is kind of off here

**Review by chouvinc — APPROVED** [review]
Seems good.
